### PR TITLE
LA-709 Support drive function

### PR DESCRIPTION
**Pull request by dab246**
- [x] Separate context menu for Drive
- [x] Open drive 

```
@GET /webservice/rest/user/v4/shared_spaces?parent={id}&withRole=true
```
- [x] Support drive destination picker

https://user-images.githubusercontent.com/80730648/141262483-a5af687c-0290-4e72-9538-bf28d91c3d39.mov

https://user-images.githubusercontent.com/80730648/141262514-d10a1edd-3c99-4fc2-bdf4-1f4e25252f8b.mov

### CODE REVIEW

File: domain/lib/src/model/sort/order_screen.dart [hunk 1]
```diff
@@ -32,6 +32,7 @@ enum OrderScreen {
   mySpace,
   sharedSpaceDocument,
   workGroup,
+  sharedSpaceDrive,
```
Comment (hoangdat) [inline domain/lib/src/model/sort/order_screen.dart:35]: insideDrive

File: domain/lib/src/repository/sharedspace/shared_space_repository.dart [hunk 2]
```diff
@@ -56,4 +56,6 @@ abstract class SharedSpaceRepository {
     SharedSpaceId sharedSpaceId,
     SharedSpaceRole sharedSpaceRole,
     EnableVersioningWorkGroupRequest enableVersioningWorkGroupRequest);
+
+  Future<List<SharedSpaceNodeNested>> getAllSharedSpaceDrive(SharedSpaceId sharedSpaceId);
```
Comment (hoangdat) [inline domain/lib/src/repository/sharedspace/shared_space_repository.dart:60]: Please change all `SharedSpace` to `WorkGroup` for only you changes

Comment (hoangdat) [inline domain/lib/src/repository/sharedspace/shared_space_repository.dart:60]: Please create `drive_repository` and `drive_datasource` for drive. Then move this function to `drive_repository`. After that, I think the method name should be `getAllWorkgroups(DrivedId driveId)`

Create `DriveId` also, then an extension `DrivedId SharedSpaceId.asDriveId()`

File: lib/presentation/widget/shared_space_document/shared_space_document_arguments.dart [hunk 3]
```diff
@@ -39,13 +39,15 @@ class SharedSpaceDocumentArguments extends RouterArguments {
   final SharedSpaceDocumentType documentType;
   final WorkGroupFolder? workGroupFolder;
   final SharedSpaceNodeNested sharedSpaceNode;
+  final SharedSpaceNodeNested? sharedSpaceDrive;
```
Comment (hoangdat) [inline lib/presentation/widget/shared_space_document/shared_space_document_arguments.dart:42]: `drive`

**Comment by hoangdat** [comment]
I propose the name: 
- `drive_widget` - `drive_view_model`: list all drive and workgroup without parent
- `workgroup_widget` - `workgroup_view_model`: inside drive, list all the workgroup of a drive 
- keep the current status for inside workgroup/sharedSpace

WDYT? @dab246 @huynguyennovem

**Comment by dab246** [comment]
> I propose the name:
> 
> * `drive_widget` - `drive_view_model`: list all drive and workgroup without parent
> * `workgroup_widget` - `workgroup_view_model`: inside drive, list all the workgroup of a drive
> * keep the current status for inside workgroup/sharedSpace
> 
> WDYT? @dab246 @huynguyennovem


IMO still keep `SharedSpaceWidget` - SharedSpaceModel` to list all drive and workgroup without parent

**Comment by hoangdat** [comment]
> > I propose the name:
> > 
> > * `drive_widget` - `drive_view_model`: list all drive and workgroup without parent
> > * `workgroup_widget` - `workgroup_view_model`: inside drive, list all the workgroup of a drive
> > * keep the current status for inside workgroup/sharedSpace
> > 
> > WDYT? @dab246 @huynguyennovem
> 
> IMO still keep `SharedSpaceWidget` - `SharedSpaceModel` to list all drive and workgroup without parent

ok, go

**Review by hoangdat — APPROVED** [review]

**Pull request merged**